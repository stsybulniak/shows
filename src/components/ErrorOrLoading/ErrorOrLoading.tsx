import { FC } from 'react';
import Error from './Error/Error';
import Loader from './Loader/Loader';

interface IErrorOrLoadingProps {
  isLoading: boolean;
  error: string | null;
}

const ErrorOrLoading: FC<IErrorOrLoadingProps> = (props) => {
  const { isLoading, error, children } = props;
  if (isLoading) {
    return <Loader />;
  }
  if (error) {
    return <Error message={error} />;
  }

  return <>{children}</>;
};

export default ErrorOrLoading;
