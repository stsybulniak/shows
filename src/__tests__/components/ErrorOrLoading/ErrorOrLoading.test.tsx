import { render } from '@testing-library/react';
import ErrorOrLoading from '../../../components/ErrorOrLoading/ErrorOrLoading';

describe('<ErrorOrLoading />', () => {
  it('ErrorOrLoading shows Loading component', () => {
    const { baseElement } = render(
      <ErrorOrLoading isLoading error={null}>
        Some child
      </ErrorOrLoading>
    );
    expect(baseElement).toMatchSnapshot();
  });

  it('ErrorOrLoading shows Error component', () => {
    const { baseElement } = render(
      <ErrorOrLoading isLoading={false} error={'Some error'}>
        Some child
      </ErrorOrLoading>
    );
    expect(baseElement).toMatchSnapshot();
  });

  it('ErrorOrLoading shows child component', () => {
    const { baseElement } = render(
      <ErrorOrLoading isLoading={false} error={null}>
        Some child
      </ErrorOrLoading>
    );
    expect(baseElement).toMatchSnapshot();
  });
});
