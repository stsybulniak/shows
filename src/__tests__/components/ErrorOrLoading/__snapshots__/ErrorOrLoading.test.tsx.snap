// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorOrLoading /> ErrorOrLoading shows Error component 1`] = `
<body>
  <div>
    <h3>
      Some error
    </h3>
  </div>
</body>
`;

exports[`<ErrorOrLoading /> ErrorOrLoading shows Loading component 1`] = `
<body>
  <div>
    <h4>
      Loading...
    </h4>
  </div>
</body>
`;

exports[`<ErrorOrLoading /> ErrorOrLoading shows child component 1`] = `
<body>
  <div>
    Some child
  </div>
</body>
`;
